### ISSUE TITLE
Special characters in pilot names

### ISSUE THREAD

**Issue by SJuliez** [MegaMek/megamek#62]
Special characters in pilot names (e.g. José) do get saved correctly, as far as I can tell, but they don't seem to get loaded back in correctly, as the tooltip and mech display show "?" or other nonsense when loading such a savegame.

**Comment by arlith** [comment]
Could you attach an example saved game?

**Comment by SJuliez** [comment]
[specialchars.sav.gz](https://github.com/MegaMek/megamek/files/160213/specialchars.sav.gz)

**Comment by Akjosch** [comment]
Quick analysis: They aren't saved correctly. Here's what the pilot's name looks like when I open it up with UTF-8 encoding (the numbers in parentheses represent illegal characters and their hexadecimal values):

    Jos(E9) Gar(F2)n(DF)

Here's the same in ISO-8859-15

    José Garònß

Encoding the names properly in UTF-8 lets MegaMek load them properly too.

[issue52_specialchars.sav.gz](https://github.com/MegaMek/megamek/files/169706/issue52_specialchars.sav.gz)

**Comment by Akjosch** [comment]
Side note: Saving (and loading, presumably) still uses XStream instead of JAXB. I have an urge to just rip it all out and replace it. This would speed it up, make the save files vastly smaller, and fix this issue along the way too. On the downside, it would make the save games incompatible - you couldn't load old ones in the newer version. Comments?

**Comment by Dylan-M** [comment]
Typically doing that is bad form anyway, I say go for it. But not until we start the new stable.

Also this goes into the new policies thing Arlith and I are still working on. Essentially any large new feature should be worked in it's own branch (So you could start work now and merge it post-stable). Criteria for "large new feature" will be something along the lines of > 1 hr work and/or more than 1 commit.

**Comment by arlith** [comment]
If someone wants to do the work of removing XStream, I'm ok with it.  Ideally, it should be easy to add serialization for new state in Game (which I think JAXB satisfies).  I'm also generally okay with breaking saved game compatibility between releases.  I don't think it's unreasonable to require people to finish up their current game before upgrading to a new version.  This is less true with MekHQ campaign files, where those persist for a long time.

**Comment by nderwin** [comment]
Ideally, you could detect the character encoding of the file before opening it, but then always save the files with UTF-8.  Over time, you could then change to only open with UTF-8 if you wanted to simplify the code, but at least that could be a non-breaking path forward.

**Comment by Akjosch** [comment]
I added a small hotfix which saves the game with the proper encoding (UTF-8). This won't fix already saved games though. @SJuliez, it's up to you to decide if this fixes your problem or not.

**Comment by SJuliez** [comment]
Perfect.
